ui/src/main/webapp/src/app/components/reports/application-index/application-index.component.ts

```diff
@@ -0,0 +1,263 @@
+    }
+
+    private getCategoryStats(category:string, incidents:ChartStatistic[], points:ChartStatistic[]): any[] {
+        if ( !category )
+            return;
+
+        let result: any[] = [];
+        let series: ChartStatistic[] = [];
+        
+        var incidentsSum = getSumOfArray(incidents);
+        series.push(new ChartStatistic("incidents",incidentsSum));
+        
+        var pointsSum = getSumOfArray(points);
+        series.push(new ChartStatistic("points", pointsSum));
+           
+        result.push({"name": category, "series": series});
+        return result;
+    }
+
+    private getCategoryDetailsStats(incidents:ChartStatistic[]): any[] {
+        let result: any[] = [];
+        
+        Object.keys(incidents).forEach( effort => {
+            let series: ChartStatistic[] = [];
+            var effortIncidents:ChartStatistic = incidents[effort];
+
+            var incidentsNumber:number = effortIncidents.value;
+            var storyPointPerIncident: number = EffortLevel[effortIncidents.name];
+
+            series.push(new ChartStatistic("incidents",incidentsNumber));
+            series.push(new ChartStatistic("points",incidentsNumber*storyPointPerIncident));
+            result.push({"name": effortIncidents.name, "series": series});
+        });
+          
+        return result;
+    }
+    private getCategoriesStats(incidents:ChartStatistic[], points:ChartStatistic[]): any[] {
+  
+        let result: any[] = [];
+
+        Object.keys(incidents).forEach(category => {
+            var categoryStr:string = incidents[category].name;
+            let series: ChartStatistic[] = [];
+            series.push(new ChartStatistic("incidents",incidents[category].value));
+            series.push(new ChartStatistic("points", points[category].value));
+            result.push({"name": categoryStr, "series": series});
+        });
+        return result;
+    }
+
+    multiTest = [
+        {"name":"mandatory", "series": [ new ChartStatistic("Incidents",5),new ChartStatistic("Points",15) ]},
+        {"name":"optional", "series": [ new ChartStatistic("Incidents",5),new ChartStatistic("Points",5) ]},
+        {"name":"potential", "series": [ new ChartStatistic("Incidents",1),new ChartStatistic("Points",3) ]}
+    ];
+}
+
+class ChartStatistic {
+    name:string;
+    value:number;
+
+    constructor(name: string, value: number) {
+        this.name = name;
+        this.value = value;
+    }
+
+    public getValue():number{
+        return this.value;
+    }
+}
+
+class MultiSeriesChartStatistic {
+    name:string;
+    series:ChartStatistic[];
+
+    constructor(name: string, value: ChartStatistic[]) {
+        this.name = name;
+        this.series = value;
+    }
+}
+
+function getSumOfArray(array: ChartStatistic[]):number{
+    var sum: number = 0;
+    var i = array.length; 
+
+    while (i--) {
+        var stat:ChartStatistic  = array[i];
+        sum = sum + stat.value;
+    }
+    return sum;
+}
```

Comment:
I know there is Array.reduce method, but for this usage, it wasn't working as there were some type cast errors like Number can't have binary operator `+`. So i changed that to more old fashion for loop.